### DISCUSSION TITLE
How to merge ArchR projects

### DISCUSSION THREAD

**Discussion by Scarlett-SYNC** [GreenleafLab/ArchR#1070] [Questions / Documentation]
Hello! 

I have two samples, and for each sample there are three replicates. For sample A, there are a1, a2, a3; for sample B, there are b1, b2, b3. For each replicates, I create a ArchR project. Now I want to compare the two samples to get differential peaks, how can I merge the six ArchR projects to form one ArchR project, which can group by sample? (Let a1+a2+a3 compare with b1+b2+b3)

Thank you!

Best regards,
Shijing

**Comment by rcorces** [comment]
As indicated when you tried to open this issue, Issues are not for usage questions. This belongs in Discussions and I have moved it there.

![image](https://user-images.githubusercontent.com/10343046/136846878-50a5e2cd-9081-4438-a611-b57317345fac.png)

**Comment by rcorces** [comment]
Your question indicates some confusion about ArchR basics. Please read the manual in full and go through the tutorial.

An ArchRProject consists of multiple arrow files. Each arrow file corresponds to a different sample. You cannot compare across ArchRProjects nor can you merge ArchRProjects. You should create a single ArchRProject containing all of the samples you wish to compare.

**Reply by brianpenghe** [reply]
What if we subset an ArchR object for subclustering and want to merge after annotation?

**Reply by rcorces** [reply]
ArchRProjects cannot be merged at the moment (v1.0.1 and v1.0.2). It is unlikely that this will become a feature anytime soon given the data structures we are currently using.

If you annotate a subset of your project, you can use the corresponding cellNames and annotations to add these back to your pre-subset project.

**Reply by brianpenghe** [reply]
That's exactly what I'm trying to do. My plan is to save the annotated metadata in data.frame and merge the data.frame variables. Then put them back into the original ArchRProject. Does it sound reasonable to you?

**Reply by rcorces** [reply]
you cant add data.frames to an ArchRProject per se - you will have to update the cellColData with additional annotation columns

**Reply by brianpenghe** [reply]
Yes it worked! Thanks!

**Reply by Zepeng-Mu** [reply]
I'm wondering whether I can run `ArchRProject` using arrow files `A` and `B` created under projects `A` and `B`, respectively.
`ArchRProject` accepts more than one arrow files, but it does not seem to work in this case, as there is an error message when I tried to subset the merged project.

**Reply by rcorces** [reply]
The problem is that ArrowFiles have a lot of information that is expected based on values stored in the ArchRProject. So it would be extremely easy to run into incompatibilities doing this. This isnt really a use case that is supported by ArchR.

**Reply by Zepeng-Mu** [reply]
OK, thanks. I should start with fragment files to make arrow files then.
